client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-name.js

```diff
@@ -12,10 +12,46 @@ import Gridicon from 'gridicons';
 import Button from 'components/button';
 import Card from 'components/card';
 import FormTextInput from 'components/forms/form-text-input';
-import { getCurrentlyEditingShippingZone } from 'woocommerce/state/ui/shipping/zones/selectors';
+import { decodeEntities } from 'lib/formatting';
 import { changeShippingZoneName } from 'woocommerce/state/ui/shipping/zones/actions';
 import { bindActionCreatorsWithSiteId } from 'woocommerce/lib/redux-utils';
 
+export const getZoneName = ( zone, locations, translate, returnEmpty = false ) => {
+	if ( zone.name ) {
+		return zone.name;
+	}
+
+	if ( returnEmpty ) {
+		return '';
+	}
+
+	if ( ! locations || ! locations.length ) {
+		return translate( 'Empty zone' );
+	}
+
+	const locationNames = locations.map( ( { name, postcodeFilter } ) => (
+		postcodeFilter ? postcodeFilter : decodeEntities( name )
```

Comment:
The `postcodeFilter` means nothing without the country name. It should be `CountryName (postcodeFilter)` or something like that.